aten/src/ATen/native/Activation.cpp

```diff
@@ -195,20 +199,37 @@ Tensor & celu_(Tensor & self, Scalar alpha) {
 }
 
 Tensor silu(const Tensor& self) {
-  return self * at::sigmoid(self);
+  Tensor result = at::native::empty_like(self, at::MemoryFormat::Contiguous);
+  at::silu_out(result, self);
+  return result;
 }
 
 Tensor& silu_(Tensor& self) {
-  return self.mul_(at::sigmoid(self));
+  return at::silu_out(self, self);
 }
 
 Tensor& silu_out(Tensor& result, const Tensor& self) {
-  return at::mul_out(result, self, at::sigmoid(self));
+  auto iter = TensorIterator::unary_op(result, self);
```

Comment:
please check that result type is same as self, so that `out` variant of the function properly errors out if user sent a wrong argument

Reply:
Done